hedera-node/hedera-token-service/src/main/java/com/hedera/node/app/service/token/entity/AccountBuilder.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.service.token.entity;
+
+import com.hedera.node.app.spi.key.HederaKey;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/** Builds an account using a builder pattern */
+public interface AccountBuilder {
```

Comment:
In which cases do we need an interface of a builder?

Reply:
I have followed POC. I think the builder will not just be an internal implementation